evm/chains/base.py

```diff
@@ -384,6 +384,15 @@ def get_vm(self, header=None):
         vm_class = self.get_vm_class_for_block_number(header.block_number)
         return vm_class(header=header, chaindb=self.chaindb)
 
+    def get_block_class_for_block_number(self, block_number):
+        """
+        Returns the Block class for the given block number.
+        """
+        # This method is needed because our VM classes cannot be pickled, so we can't use
```

Comment:
It looks like if we change our VM classes to use standard inheritance they become pickleable.  cc @cburgdorf and link to #622 

```python
class FrontierVM(VM):
    # fork name
    fork='frontier'
    # classes
    _state_class = FrontierState
    # helpers
    create_header_from_parent = staticmethod(create_frontier_header_from_parent)
    compute_difficulty = staticmethod(compute_frontier_difficulty)
    configure_header = configure_frontier_header
    make_receipt = staticmethod(make_frontier_receipt)
```